blockstack/stacks-wallet-web
# feat: fee estimate ui

> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/1469040146).

This PR adds ui work and a new query for fee estimations. While implementing this I tried to do as much refactoring as I could based on our code conventions. I am sure there are things I missed in moved files, so feel free to mark changes I should make even if doesn't pertain to new code for fee estimations. I'd like to address as much as possible with each PR.

Issue #1846 

## Testing:
- Check that all transactions work correctly in the test-app using the new fee estimations
- Check that sending STX tokens still works correctly using the new fee estimations
- Fee estimations should default (load) to `Standard` (middle) estimation
- Verify that setting a custom fee works above lowest estimation
- Verity low fee warning appear on submit of tx with custom fee set lower than lowest estimate
- Check that the edit nonce settings drawer still works as expected
- Check that the increase fee settings still works as expected

*Note: It is expected that the estimations will only be slightly different. At times on testnet the middle and high estimations were the same for me.


![Screen Shot 2021-11-03 at 7 52 28 PM](https://user-images.githubusercontent.com/6493321/140239345-13ad9041-e2da-41ef-b873-011a5b365d72.png)

![Screen Shot 2021-11-03 at 7 52 46 PM](https://user-images.githubusercontent.com/6493321/140239354-7230501d-2418-4b92-b836-1cbdfea3484e.png)

![Screen Shot 2021-11-03 at 7 52 59 PM](https://user-images.githubusercontent.com/6493321/140239364-515d7171-ecbb-42c9-b6d5-3a23911562a8.png)

![Screen Shot 2021-11-03 at 7 53 30 PM](https://user-images.githubusercontent.com/6493321/140239374-b60c4d9b-bdec-4cc2-95b2-fbb457c0c242.png)

![Screen Shot 2021-11-03 at 7 53 54 PM](https://user-images.githubusercontent.com/6493321/140239398-cc76836c-b719-4bd0-ba19-734ec175dd51.png)

cc/ @kyranjamie @beguene
